### PR TITLE
Turn warning into error: transform input artifact of type File

### PR DESCRIPTION
**Pull request by lptr**
Turn this warning into an error: https://github.com/gradle/gradle/blob/23e1c4216cfb9b185fd161543f1e38844fb2a800/subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultTransformer.java#L428-L436

See https://github.com/gradle/gradle/issues/21648

### CODE REVIEW

**Comment by lptr** [comment]
@bot-gradle test QFL please.

File: subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformValuesInjectionIntegrationTest.groovy [hunk 1]
```diff
@@ -100,20 +100,63 @@ class ArtifactTransformValuesInjectionIntegrationTest extends AbstractDependency
...
         outputContains("processing b.jar")
         outputContains("processing c.jar")
         outputContains("result = [b.jar.green, c.jar.green]")
+    }
 
-        where:
-        inputArtifactType              | convertToFile   | expectedDeprecation
-        'File'                         | ''              | "Injecting the input artifact of a transform as a File has been deprecated. This will fail with an error in Gradle 8.0. Declare the input artifact as Provider<FileSystemLocation> instead. See https://docs.gradle.org/current/userguide/artifact_transforms.html#sec:implementing-artifact-transforms for more details."
-        'Provider<FileSystemLocation>' | '.get().asFile' | null
+    def "transform fails when input artifact if of type File"() {
```
Comment (alllex) [inline subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformValuesInjectionIntegrationTest.groovy:111]: 💄 Typo: should be "is of type" ?

**Review by alllex — APPROVED** [review]
LGTM!

File: subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformValuesInjectionIntegrationTest.groovy (hunk 1 above)
Comment (lptr) [reply subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformValuesInjectionIntegrationTest.groovy]: Fixed, thanks.

**Comment by lptr** [comment]
@bot-gradle test QFL please.

**Comment by blindpirate** [comment]
@bot-gradle test and merge please.

**Pull request merged**